warp10/src/main/java/io/warp10/continuum/gts/INTERPOLATE.java

```diff
@@ -27,23 +25,514 @@
+          nElevations++;
+        }
+
+        if ((null != filled.locations) && (GeoTimeSerie.NO_LOCATION != filled.locations[i])) {
+          nLocations++;
+        }
+      }
+
+      // fill occurrence ticks
+      for (Long tick: (List<Long>) params.get(PARAM_OCCURRENCES)) {
+        if (function.isValidPoint(tick)) {
+          GTSHelper.setValue(filled, tick, function.value(tick));
+        } else {
+          Number filler = (Number) params.get(PARAM_INVALID_TICK_VALUE);
+          if (null != filler) {
+            GTSHelper.setValue(filled, tick, filler);
+          }
+        }
+      }
+    }
+
+    //
+    // Take care of geo if we have at least two valid values for either elevation or location
+    //
+
+    PolynomialSplineFunction elevFunction = null;
+    PolynomialSplineFunction latFunction = null;
+    PolynomialSplineFunction lonFunction = null;
+
+    //
+    // Compute interpolators
+    //
+
+    double[] xelev = null;
+    double[] felev = null;
+    double[] xloc = null;
+    double[] flat = null;
+    double[] flon = null;
+
+    if (nElevations >= 2) {
+      xelev = new double[nElevations];
+      felev = new double[nElevations];
+    }
+
+    if (nLocations >= 2) {
+      xloc = new double[nLocations];
+      flat = new double[nLocations];
+      flon = new double[nLocations];
+    }
+
+    for (int i = 0; i < nvalues; i++) {
+      if (nElevations >= 2) {
+        if (GeoTimeSerie.NO_ELEVATION != filled.elevations[i]) {
+          xelev[i] = ((Number) GTSHelper.tickAtIndex(filled, i)).doubleValue();
+          felev[i] = ((Number) filled.elevations[i]).doubleValue();
+        }
+      }
+
+      if (nLocations >= 2) {
+        if (GeoTimeSerie.NO_LOCATION != filled.locations[i]) {
+          xloc[i] = ((Number) GTSHelper.tickAtIndex(filled, i)).doubleValue();
+
+          double[] latlon = GeoXPLib.fromGeoXPPoint(filled.locations[i]);
+          flat[i] = latlon[0];
+          flon[i] = latlon[1];
+        }
+      }
+    }
+
+    //
+    // Fill elevation
+    //
+
+    if (nElevations >= 2) {
+      if (null == params.get(PARAM_INTERPOLATOR_GEO)) {
+        elevFunction = (new LinearInterpolator()).interpolate(xelev, felev);
+      } else {
+        switch (Interpolator.valueOf((String) params.get(PARAM_INTERPOLATOR_GEO))) {
+          case spline:
+            if (nElevations > 2) {
+              elevFunction = (new SplineInterpolator().interpolate(xelev, felev));
+              break;
+            }
+          case akima:
+            if (nElevations > 4) {
+              elevFunction = (new AkimaSplineInterpolator().interpolate(xelev, felev));
+              break;
+            }
+          case linear:
+            elevFunction = (new LinearInterpolator()).interpolate(xelev, felev);
+            break;
+          case noop:
+            elevFunction = null;
+            break;
+        }
+      }
+
+      //
+      // Sort ticks
+      //
+
+      GTSHelper.sort(filled);
+
```

Comment:
This is could be nice, but is not in the logic of original `INTERPOLATE` function. It adds all the new datapoints after the existing ones.

Reply:
this is done only if some geo interpolation is perform (like for the original implementation)